midterm/src/Directory.hs

```diff
@@ -1,17 +1,48 @@
-module Directory where
+module Directory (DirectoryTree(..), buildTree, defaultDisplayTree, alphabetisedDisplayTree) where
 
-import Tree
+import System.Directory (doesDirectoryExist, listDirectory)
+import System.FilePath ((</>))
+import Data.List (sort)
 
 -- Define an appropriate type called `DirectoryTree` to represent the directory hierarchy
+data DirectoryTree = File FilePath | Directory FilePath [DirectoryTree]
+    deriving (Show, Eq)
 
 -- Create a tree that represents the hierarchy of files and subdirectories of the given directory.
 buildTree :: FilePath -> IO DirectoryTree
-buildTree = undefined 
+buildTree filePath = do
+    check <- doesDirectoryExist filePath
+    if check
+        then do
+            files <- listDirectory filePath
+            let sortedFiles = sort files -- probably sorting by default is more efficient than separating by default(???)
```

Comment:
Sorting makes zero sense here because there are two display functions, both of which impose their own order.

Reply:
Is it really makes zero sense?
I thought it would be more efficient: we don’t have to sort it every time we display it, because usually we need to build it one time, and then multiple times to display it.
Also I supposed that alphabetical order sorting is longer than separating(NLogN vs N), so I actually pre-calculated it here